pytorch_lightning/core/lightning.py

```diff
@@ -1729,6 +1730,36 @@ def to_onnx(self, file_path: str, input_sample: Optional[Tensor] = None, **kwarg
 
         torch.onnx.export(self, input_data, file_path, **kwargs)
 
+    def to_torchscript(self) -> torch.jit.ScriptModule:
+        """
+        Compiles the model to a :class:`~torch.jit.ScriptModule`.
+        This can be overridden to support custom TorchScript module export.
+
+        Note:
+            - Requires the implementation of the :meth:`LightningModule.forward` method.
+            - It is recommended that you install the latest version of PyTorch to use this feature without limitations.
+
+
+        Example:
+            >>> class SimpleModel(LightningModule):
+            ...     def __init__(self):
+            ...         super().__init__()
+            ...         self.l1 = torch.nn.Linear(in_features=64, out_features=4)
+            ...
+            ...     def forward(self, x):
+            ...         return torch.relu(self.l1(x.view(x.size(0), -1)))
+            ...
+            >>> model = SimpleModel()
+            >>> torch.jit.save(model.to_torchscript(), "model.pt")  # doctest: +SKIP
```

Comment:
why skipping these?

Reply:
torchscript needs the source file to work, but in doctests it doesn't seem to be working, errors with something like "can't find source file". so I kept it for documentation, as an example. I know it's a bit dumb... better to remove? @justusschock

Reply:
No, was just wondering why it did not work. Think they should stay as examples.

Reply:
maybe add a comment about the skip to the code